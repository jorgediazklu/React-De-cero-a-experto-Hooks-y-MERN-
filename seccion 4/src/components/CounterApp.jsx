import React, { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);
  return (
    <div>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={() => setCounter((prev) => prev + 1)}>+1</button>
      <button onClick={() => setCounter((prev) => prev - 1)}>-1</button>
      <button onClick={() => setCounter(value)}> reset</button>
    </div>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
