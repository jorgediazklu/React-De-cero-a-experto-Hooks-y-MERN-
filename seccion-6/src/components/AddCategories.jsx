import React, { useState } from "react";

export const AddCategories = ({ onAddCategory }) => {
  const [inputValue, setInputValue] = useState("");
  const handleChange = (e) => {
    setInputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onAddCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        onChange={handleChange}
        placeholder="Buscar gifts"
        value={inputValue}
      />
    </form>
  );
};
