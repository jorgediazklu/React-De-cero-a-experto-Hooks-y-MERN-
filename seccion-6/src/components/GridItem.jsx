import React from "react";

export const GridItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url.url} alt="" />
      <p>{title}</p>
    </div>
  );
};
