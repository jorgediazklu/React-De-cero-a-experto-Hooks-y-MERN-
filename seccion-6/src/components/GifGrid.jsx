import React, { useEffect, useState } from "react";
import { GridItem } from "./GridItem";
import { useFetchGifs } from "../hooks";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <div>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando...</h2>}
      <div className="card-grid">
        {images.map(
          (gif) => gif.title.length > 0 && <GridItem key={gif.id} {...gif} />
        )}
      </div>
    </div>
  );
};
