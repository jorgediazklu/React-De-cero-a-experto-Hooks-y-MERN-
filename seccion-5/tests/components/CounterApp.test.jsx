import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../../src/components/CounterApp";

describe("CounterApp.test", () => {
  const value = 100;

  test("<CounterApp/> match snapshot", () => {
    const { container } = render(<CounterApp value={value} />);
    expect(container).toMatchSnapshot();
  });

  test("<CounterApp/> match snapshot", () => {
    const { getByText } = render(<CounterApp value={value} />);
    expect(getByText(value).innerHTML).toEqual("100");
  });

  test("incrementar con el boton +1 ", () => {
    render(<CounterApp value={100} />);
    //fireEvent: ejecutar eventos
    fireEvent.click(screen.getByText("+1"));
    //espera que alla incrementado
    expect(screen.getByText("101")).toBeTruthy();
  });

  test("incrementar con el boton -1 ", () => {
    render(<CounterApp value={100} />);
    //fireEvent: ejecutar eventos
    fireEvent.click(screen.getByText("-1"));
    //espera que alla incrementado
    expect(screen.getByText("99")).toBeTruthy();
  });
  test("resetear", () => {
    render(<CounterApp value={355} />);
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("reset"));
    expect(screen.getByText(355)).toBeTruthy();
  });
});
