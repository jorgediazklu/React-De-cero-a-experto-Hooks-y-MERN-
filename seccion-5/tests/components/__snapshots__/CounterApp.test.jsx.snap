// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp.test <CounterApp/> match snapshot 1`] = `
<div>
  <div>
    <h1>
      CounterApp
    </h1>
    <h2>
      100
    </h2>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button>
       reset
    </button>
  </div>
</div>
`;
